### ISSUE TITLE
UI - value is missing in the request when installing Monitoring v2 with persistent storage for Grafana 

### ISSUE THREAD

**Issue by benzman81** [rancher/dashboard#1689]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Install rancher-monitoring with selected storage for grafana using `Statefulset template.` 

**Result:**
No volume will be used for grafana.

**Other details that may be helpful:**
When you switch to edit the yaml before install, you can see, that the setting `persistence.enabled: true` is missing for grafana. Manually adding this will create the volume, but also it will not show to be selected in the rancher form when you try to upgrade.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.5.1/k8s 1.19.3
- Installation option (single install/HA):
Single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Nodes have version from 17.9.1, 18.9.0 and 19.3.5
Rancher server is docker version 17.03.2-ce
```

**Comment by jiaqiluo** [comment]
The 1st part of the issue is solved: the value `enabled: true` is sent in the API request and the volume is created and used by Grafana. 
![image](https://user-images.githubusercontent.com/6218999/97615183-36ec0580-19d8-11eb-9227-67f15aee6767.png)

But the 2nd part of the issue still exists: when editing the monitoring app, the form shows persistent storage is disabled when it is actually enabled. 

![image](https://user-images.githubusercontent.com/6218999/97615672-da3d1a80-19d8-11eb-8347-fab27f64bbe5.png)

**Comment by jiaqiluo** [comment]
The fix is valdated on rancher:`v2.5-94397a0c5f79394e0e0f360e011593110934ddec-head` with the latest UI

the 2nd part of the issue is fixed: now when editing the monitoring app as a form, the form shows the correct configuration for the persistent storage for Grafana. 

![image](https://user-images.githubusercontent.com/6218999/97896774-c81ded80-1cf2-11eb-8653-2cd9a78ace91.png)
